### ISSUE TITLE
Refactor JSON parser

### ISSUE THREAD

**Issue by nicoweidner** [spdx/tools-python#305]
We want a refactored parser using the new data model (available on https://github.com/spdx/tools-python/tree/refactor-python-tools).

Possible basic interface (maybe extract an abstract base class later) as follows. The parser class should use composition instead of inheritance.
```
class JsonParser:
    creation_info_parser: CreationInfoParser
    package_parser: PackageParser
    file_parser: FileParser
    <...>

    def parse(self, filename: str) -> Document:
        logger = Logger()
         with open(filename) as file:
             input_doc_as_dict = json.load(file)
         document: Document = creation_info_parser.parse(input_doc_as_dict, logger)
         document.packages = package_parser.parse(input_doc_as_dict.get("packages"), logger)
        <...>
        
        if logger.has_errors():
            raise SpdxParsingError(logger.get_errors())
        return document
```

**Comment by nicoweidner** [comment]
Given the changes in https://github.com/spdx/tools-python/issues/306, we should adapt the structure slightly: `creation_info_parser` will return a `CreationInfo` object that we can use to instantiate `Document`

**Comment by meretp** [comment]
fixed with #366
